crates/rpc/rpc-engine-api/src/engine_api.rs

```diff
@@ -265,13 +265,24 @@ where
         )?;
 
         // Now resolve the payload
-        Ok(self
+        let mut resolved_payload = self
             .inner
             .payload_store
             .resolve(payload_id)
             .await
             .ok_or(EngineApiError::UnknownPayload)?
-            .map(|payload| payload.into_v3_payload())?)
+            .map(|payload| payload.into_v3_payload())?;
+
+        // After `Cancun` is enabled on optimism, an extra field `parent_beacon_block_root` is
```

Comment:
I'm okay with this for now, but ideally this should be solved with the Op payload type instead